.github/workflows/measureMemory.yml

```diff
@@ -0,0 +1,412 @@
+# This is a basic workflow to help you get started with Actions
+
+name: Measure memory
+
+# Controls when the action will run. Triggers the workflow push is only on main branch and PR on any branch.
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - "*"
+  pull_request:
+  workflow_dispatch:
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # This workflow contains a single job called "build"
+  generateSummary:
+    needs: coapGateway
+    runs-on: ubuntu-latest
+    steps:
+      - name: Download artifacts
+        uses: actions/download-artifact@v3
+        with:
+          path: outputs
+      - name: Update summary
+        run: |
+          echo "### CoAP Gateway memory usage" >> $GITHUB_STEP_SUMMARY
+          echo "" >> $GITHUB_STEP_SUMMARY
+          echo "The following table displays the memory usage of the coap-gateway when connected to varying numbers of devices, resources, and resource sizes. One by one, the devices were connected and the following actions were executed for each device:" >> $GITHUB_STEP_SUMMARY
+          echo "" >> $GITHUB_STEP_SUMMARY
+          echo "- Sign up" >> $GITHUB_STEP_SUMMARY
+          echo "- Sign in" >> $GITHUB_STEP_SUMMARY
+          echo "- Publish resources" >> $GITHUB_STEP_SUMMARY
+          echo "- Synchronize with twin device" >> $GITHUB_STEP_SUMMARY
+          echo "- Ping until test ends" >> $GITHUB_STEP_SUMMARY
+          echo "" >> $GITHUB_STEP_SUMMARY
+          echo "| Num devices | Num Resources | Resource Data Size(KB) | Current memory usage (MB) | Duration (seconds) |" >> $GITHUB_STEP_SUMMARY
+          echo "| ----------- | ------------- | ------------------ | ------------------------- | -------- |" >> $GITHUB_STEP_SUMMARY
+          jq -s '.|=sort_by(.NumDevices,.NumResources,.LogLevel,.LogDumpBody,.ResourceDataSize)' outputs/*/*.json > output.json
+          for row in $(cat output.json | jq -r '.[] | @base64'); do
+            _jq() {
+              echo ${row} | base64 --decode | jq -r ${1}
+            }
+            duration=$(echo "scale=2 ; $(_jq '.Duration') / 1000000000" | bc )
+            resourceDataSize=$(echo "scale=2 ; $(_jq '.ResourceDataSize') / 1024" | bc )
+            echo "| $(_jq '.NumDevices') | $(_jq '.NumResources') | $resourceDataSize | $(_jq '.CurrentMemRSS') | $duration |" >> $GITHUB_STEP_SUMMARY
+          done
+          
+  coapGateway:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          #1
+          - name: devices/1/resources/1/size/16KB/debug/dumpBody
+            numDevices: 1
+            numResources: 1
+            resourceDataSize: 16384
+            expectedRSSInMB: 35
+            logLevel: info
```

Comment:
`logLevel` and `logDumpBody` have always the same value, so they could be hard coded in the `make test/mem` invocation on line 398

Reply:
removed